File: stack.cpp
#include<iostream>
using namespace std;
template <class t> 
class stack{
	int top;
	t item[100000];
public:
	stack()
	{
		top=0;
	}
	void push(t x)
	{
		if(top>=100000)
		cout<<"No Place ";
		else
		{
		item[top]=x;
		top++;
		}
	}
	void pop()
	{
		if(top==0)
		{
			cout<<"No Element";
		}
		else 
		top--;
	}
	int get_top()
	{
		return item[top-1];
	}
	void print ()
	{
		int te=top;
		for(int i=te-1;i>=0;i--)
		{
			cout<<item[i]<<" ";
		}
		cout<<endl;
	}
		
};
int main()
{
	stack <int>s;
	s.push(10);s.push(15);s.push(20);
	s.print();
	cout<<s.get_top()<<"  ";
}

File: stack.java
package com.company;
import java.util.EmptyStackException;
public class stack
{
public class node
{
    public int item;
    public node next;
    node()
    {
        next=null;
    }
}
node top=null;
public void push(int it)
{
   node temp=new node();
 temp.item=it;
 temp.next=top;
 top=temp;
}
public boolean empty()
{
    if(top==null)
        return true;
    else
        return false;
}
public  void pop()
{
    node templ=new node();
    templ=top;
     top=top.next;
     templ.next=null;


}
    public void display()
    {
        if (top == null) {
            System.out.printf("\nStack Underflow");

        }
        else {
            node temp = top;
            while (temp != null)
            {
                System.out.printf("%d->", temp.item);
                temp = temp.next;
            }
        }
    }

}